### PR TITLE
Support new ECMAScript operators

### PR DESCRIPTION
**Pull request by foxtree**
Support for the "Optional chaining" operator, in particular, should result a lot more concise and readable code.

Both of these appear to be "Stage 4" by TC-39 -- the final phase such that they recommended for adoption by implementors: 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -11,6 +11,7 @@
 !/src/**
 !package.json
 !package-lock.json
+# !.babelrc
 !config-overrides.js
```
Comment (foxtree) [inline .dockerignore:15]: comment for future reference

File: config-overrides.js [hunk 2]
```diff
@@ -1,4 +1,35 @@
...
+  // adjustStyleLoaders,
+  // addPostcssPlugins,
+  addBabelPlugins,
+  // addBabelPlugin,
+  // addExternalBabelPlugins,
+  // addExternalBabelPlugin,
+  // addDecoratorsLegacy,
+  // fixBabelImports,
+  // useBabelRc,
+} = require('customize-cra');
```
Comment (foxtree) [inline config-overrides.js:15]: comments are for future reference, in case this config needs to be revisited, again

File: config-overrides.js [hunk 3]
```diff
@@ -1,4 +1,35 @@
...
+// const postCssNested = require('postcss-nested');
+
+// Documentation for operators:
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining
+
+module.exports = (config, env) => {
```
Comment (gilhsu) [inline config-overrides.js:22]: cleanup per Github notes below?

File: config-overrides.js [hunk 4]
```diff
@@ -1,4 +1,35 @@
-module.exports = (config) => {
-  require('react-app-rewire-postcss')(config, true); // eslint-disable-line global-require
+const rewirePostCSS = require('react-app-rewire-postcss');
```
Comment (gilhsu) [inline config-overrides.js:1]: should this be `import`? 

also, should it be updated to dependences per Github note below?

File: src/components/DiscoveryApp/Api.js [hunk 5]
```diff
@@ -90,7 +90,7 @@ const itemDate = (item, category) => {
 // See queryOptions()
 const isCategory = (input, value) => {
   //      console.log('input.category: ' + JSON.stringify(input.category,null,3));
-  if (input && input.category) {
+  if (input?.category) {
     if (input.category.text === value) {
```
Comment (gilhsu) [inline src/components/DiscoveryApp/Api.js:94]: nice

**Review by gilhsu — APPROVED** [review]

File: config-overrides.js (hunk 4 above)
Comment (foxtree) [reply config-overrides.js:1]: This is executing in a Node.js context, so no.  (no `import` as of Node v14?) 

The issue is that ESLint does not recognize this a Node.js file -- but that is configurable.

File: config-overrides.js (hunk 3 above)
Comment (foxtree) [reply config-overrides.js:22]: I think all these warnings are a result of `.github/workflows/eslint.yml` not suppressing warnings.     (Not enough time to fix everything in the legacy codebase, except that which is auto-fixable.)  



some of these warnings are also irrelevant, e.g.:
> `'postcss-nested' should be listed in the project's dependencies, not devDependencies`


 but the documentation for `postcss-nested` says to install it via `devDependencies`:

image

So:  the ESLint config could be tweaked, or, all warnings suppressed in the CI task...

Comment (foxtree) [reply config-overrides.js:22]: ...I've added a commit to suppress ESLint warnings in CI.

**Pull request merged**